presto-main/src/main/java/io/prestosql/operator/LookupJoinPageBuilder.java

```diff
@@ -179,7 +194,7 @@ private void appendProbeIndex(JoinProbe probe)
         for (int index : probe.getOutputChannels()) {
             Block block = probe.getPage().getBlock(index);
             // Estimate the size of the current row
-            estimatedProbeBlockBytes += block.getSizeInBytes() / block.getPositionCount();
+            estimatedProbeBlockBytes += block.isLoaded() ? block.getSizeInBytes() / block.getPositionCount() : 0;
```

Comment:
don't use `?` condition, but rather explicit `if`.

Making `estimatedProbeBlockBytes` 0 means we might produce too large pages when they are materialized.

Please add a check in `io.prestosql.operator.LookupJoinPageBuilder#isFull` that the number of positions do not exceed `PageProcessor#MAX_BATCH_SIZE`.

Please add a `TODO` that we should estimate probe block bytes for lazy pages too.